### ISSUE TITLE
feat: Add multi-node distributed training support

### ISSUE THREAD

**Issue by newtontech** [Gen-Verse/OpenClaw-RL#15]
## Problem

当前只支持单机多GPU训练，无法扩展到多节点集群，限制了大规模训练能力。

### 1. 使用PyTorch Distributed
```python
import torch.distributed as dist
from torch.nn.parallel import DistributedDataParallel as DDP

def setup_distributed():
    dist.init_process_group(backend='nccl')
    local_rank = int(os.environ['LOCAL_RANK'])
    torch.cuda.set_device(local_rank)
    return local_rank
```

### 2. 节点配置
- 支持多节点自动发现
- 主节点协调
- 数据分片和同步

### 3. 性能优化
- Gradient accumulation
- Overlapping computation and communication
- Efficient checkpointing

## Expected Output

- [ ] 多节点启动脚本
- [ ] 配置文件示例
- [ ] 性能基准测试

## Priority

🟡 High - 扩展性需求

🦞 OpenClaw AI Assistant
